app/assets/javascripts/darkswarm/services/cart.js.coffee

```diff
@@ -61,14 +61,16 @@ Darkswarm.factory 'Cart', (CurrentOrder, Variants, $timeout, $http, $modal, $roo
       #       is unnecessary.
 
       for li in @line_items when li.quantity > 0
-        if stockLevels[li.variant.id]?
-          li.variant.on_hand = stockLevels[li.variant.id].on_hand
-          if li.quantity > li.variant.on_hand
-            li.quantity = li.variant.on_hand
-            scope.variants.push li.variant
-          if li.variant.on_hand == 0 && li.max_quantity > li.variant.on_hand
-            li.max_quantity = li.variant.on_hand
-            scope.variants.push(li.variant) unless li.variant in scope.variants
+        continue unless stockLevels[li.variant.id]?
+        li.variant.on_hand = stockLevels[li.variant.id].on_hand
+        li.variant.on_demand = stockLevels[li.variant.id].on_demand
+        continue if li.variant.on_demand
+        if li.quantity > li.variant.on_hand
+          li.quantity = li.variant.on_hand
+          scope.variants.push li.variant
+        if li.variant.on_hand == 0 && li.max_quantity > li.variant.on_hand
+          li.max_quantity = li.variant.on_hand
+          scope.variants.push(li.variant) unless li.variant in scope.variants
```

Comment:
can we add add some blank lines in between? It's feel like a wall 😱

Reply:
I have added some line breaks... I dont want to change this logic but it sure is calling for some improvement...

Reply:
not changing but a bit of spacing would ease the reading...

Reply:
Ive added some spaces, I am not sure if you are ok with it now or if you are asking for more spaces? :-D